src/index.ts

```diff
@@ -1 +1,64 @@
-console.log('hey!');
+import got from 'got';
+
+import { reject, isNil, join } from 'ramda';
+
+import { ISale } from './types';
+
+const withoutUndefined = reject(isNil);
+const apiUrl = 'https://app.enotas.com.br/api';
```

Comment:
I think a better approach here is to create a `api.ts` and inside it create a factory. Something like:

```typescript
const API_URL = 'https://app.enotas.com.br/api';

const ApiFactory = ({ apiKey, baseUrl }: { readonly apiKey: string } = { apiKey: '', baseUrl = API_URL }) => got.extend({
    baseUrl,
    headers: {
      Authorization: `Basic ${apiKey}`
    },
    json: true
  });
```

It's more functional create a `sale.ts` and inside it move `ISaleParams` , `ISale` type and this guys:
```typescript
export const getSales = (api) => ({ filter, pageNumber, pageSize, orderBy }: ISalesParams) =>  {
  const query = withoutUndefined({
        filter,
        pageNumber,
        pageSize,
        orderBy
      });

      return api.get('/vendas/getFilterBy', { query });
}

Reply:
Nice suggestion! Could you explain in more detail to @VictorOF and me why this approach would be preferable over the current approach? What are the benefits of this solution etc.?

Reply:
Sure! In resume (and in my opinion) functional programming are based in 5 easy points to remember:

- Every function need to have a return
- Functions can't edit outside states 
- Functions should make one work if you need that a function makes more than one thing you are broking the paradigma of composing stuff
- Avoid to edit variables, keep it immutable 
- Avoid to use class structures and think everything as raw data structures. Data is only data!

The actual approach constructs the factory build a client, prepare data and return an object with 3 methods (class-based structure, make more than one thing on a function, isn't compostable, the data structure isn't only a data structure).